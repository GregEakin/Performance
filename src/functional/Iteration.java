package functional;

import org.approvaltests.Approvals;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * From chapter two.
 * Created by Greg on 11/16/2014.
 */
public class Iteration {
    private static final List<String> friends = Arrays.asList("Brian", "Nate", "Raju", "Sara", "Scott");

    @Test
    public void sample1() throws Exception {
        final List<String> uppercaseNames = new ArrayList<>();
        for (String name : friends) {
            uppercaseNames.add(name.toUpperCase());
        }

        Approvals.verify(uppercaseNames.toString());
    }

    @Test
    public void sample2() throws Exception {
        final List<String> uppercaseNames = new ArrayList<>();
        for (String name : friends) {
            uppercaseNames.add(name.toUpperCase());
        }

        Approvals.verify(uppercaseNames.toString());
    }
}
