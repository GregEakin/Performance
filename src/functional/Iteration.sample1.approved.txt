[BRIAN, NATE, RAJU, SARA, SCOTT]